### ISSUE TITLE
[Accessibility]: History panel view: No indicator that panel has opened

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35924]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34447

### Problem description

1. Click this
<img width="326" alt="image" src="https://user-images.githubusercontent.com/9516420/170003365-387cb75e-8407-4731-b4f9-3156433aedf7.png">

2. View panel
<img width="1727" alt="image" src="https://user-images.githubusercontent.com/9516420/170003583-5289bd07-4781-4d0d-a918-7690561589fc.png">

Problem: There is no indication to the screen reader that anything happened.

### Expected behavior

We should:

- Shift focus to the panel when this button is clicked
- Treat this like a dialog (needs investigation), so the screen reader is aware of the updated state of the page.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
